src/Trees/Helpers.hs

```diff
@@ -163,10 +165,19 @@ bothKids :: SynTree o a -> (SynTree o a, SynTree o a)
 bothKids (Binary _ l r) = (l, r)
 bothKids _ = error "impossible to land here"
 
+swapKids :: SynTree o a -> SynTree o a
+swapKids (Binary x l r) = Binary x r l
+swapKids _ = error "impossible to land here"
+
 collectUniqueBinOpsInSynTree :: SynTree BinOp a -> [BinOp]
 collectUniqueBinOpsInSynTree (Leaf _) = []
 collectUniqueBinOpsInSynTree (Not x) = collectUniqueBinOpsInSynTree x
 collectUniqueBinOpsInSynTree (Binary op l r) = nubOrd $ concat [
   [op],
   collectUniqueBinOpsInSynTree l,
   collectUniqueBinOpsInSynTree r]
+
+mirrorTree :: SynTree o c -> SynTree o c
+mirrorTree (Binary b l r) = Binary b r l
```

Comment:
I think #29 should be completed and merged first, and then this line changed to flip the directed arrows around.